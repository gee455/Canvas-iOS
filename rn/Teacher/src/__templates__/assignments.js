//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: 1376,
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment()],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
