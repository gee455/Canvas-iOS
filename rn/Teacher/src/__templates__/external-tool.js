//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const externalTool: Template<AddressBookResult> = template({
  id: '1',
  name: 'LTI FTW',
  url: 'https://canvas.instructure.com/lti',
})

export const ltiLaunchDefinition: Template<LtiLaunchDefinition> = template({
  definition_id: 42,
  placements: {
    course_navigation: {
      url: 'https://rollcall.instructure.com/launchy-launch',
    },
  },
})
