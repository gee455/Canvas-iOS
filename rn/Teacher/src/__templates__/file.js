//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const usageRights: Template<UsageRights> = template({
  legal_copyright: '',
  use_justification: 'own_copyright',
})

export const file: Template<File> = template({
  id: '111',
  display_name: 'Book Report',
  url: 'http://canvaslms.com/bookreport',
  size: 488212,
  thumbnail_url: 'http://fillmurray.com/322/200',
  mime_class: 'document',
  preview_url: '/',
  filename: '/bookreport',
  parent_folder_id: null,
  locked: false,
  hidden: false,
  unlock_at: null,
  lock_at: null,
  usage_rights: usageRights(),
  'content-type': 'application/document',
})
