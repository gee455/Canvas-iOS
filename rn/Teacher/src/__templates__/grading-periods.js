//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const gradingPeriod: Template<GradingPeriod> = template({
  id: '1023',
  title: 'First Block',
  start_date: '2014-01-07T15:04:00Z',
  end_date: '2014-05-07T17:07:00Z',
  close_date: '2014-06-07T17:07:00Z',
  weight: 33.33,
})
