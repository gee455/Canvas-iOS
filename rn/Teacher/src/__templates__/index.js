//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './folder'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './page'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './toDo'
export * from './users'
