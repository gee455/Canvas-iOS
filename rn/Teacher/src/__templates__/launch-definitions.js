//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../utils/template'

export const launchDefinitionGlobalNavigationItem: Template<ExternalToolLaunchDefinitionGlobalNavigationItem> = template({
  message_type: 'basic-lti-launch-request',
  url: 'https://mobiledev.beta.instructuremedia.com/lti/launch?custom_gauge_launch_type=global_nav',
  title: 'Gauge',
})

export const launchDefinition: Template<ExternalToolLaunchDefinition> = template({
  definition_type: 'ContextExternalTool',
  definition_id: '360860',
  name: 'Gauge',
  description: 'Tools for Education',
  domain: 'gauge.instructure.com',
  placements: {
    global_navigation: launchDefinitionGlobalNavigationItem(),
  },
})
