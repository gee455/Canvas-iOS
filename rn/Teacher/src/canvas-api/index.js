//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import 'babel-polyfill'

export { default } from './apis/index'
export * from './apis/assignmentGroups'
export * from './apis/assignments'
export * from './apis/conversations'
export * from './apis/courses'
export * from './apis/discussions'
export * from './apis/enrollments'
export * from './apis/external-tools'
export * from './apis/groups'
export * from './apis/login'
export * from './apis/quizzes'
export * from './apis/submissions'
export * from './apis/users'
export * from './apis/media-comments'
export * from './apis/file-uploads'
export * from './apis/files'
export * from './apis/accounts'

export * from './session'

export { default as httpClient, isAbort, httpCache } from './httpClient'
