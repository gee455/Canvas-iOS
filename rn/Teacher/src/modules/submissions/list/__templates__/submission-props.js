//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
