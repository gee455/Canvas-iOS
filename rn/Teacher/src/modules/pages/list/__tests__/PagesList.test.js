//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import { shallow } from 'enzyme'
import React from 'react'
import { alertError } from '../../../../redux/middleware/error-handler'
import { httpCache } from '../../../../canvas-api/model-api'
import * as template from '../../../../__templates__'
import app from '../../../app'
import ConnectedPagesList, { PagesList } from '../PagesList'

jest.mock('../../../../redux/middleware/error-handler', () => {
  return { alertError: jest.fn() }
})
jest.useFakeTimers()

const diveList = (list: any) =>
  shallow(
    <list>
      {list.prop('data').map((item, index) =>
        <item key={list.prop('keyExtractor')(item)}>
          {list.prop('renderItem')({ item, index })}
        </item>
      )}
    </list>
  )

describe('PagesList', () => {
  let props
  beforeEach(() => {
    jest.clearAllMocks()
    httpCache.clear()
    props = {
      courseID: '1',
      navigator: template.navigator(),
      courseColor: 'green',
      course: template.courseModel({ id: '1' }),
      pages: [template.pageModel()],
      isLoading: false,
      loadError: null,
      refresh: jest.fn(),
    }
  })

  it('gets courseColor, course, and pages from the model api', () => {
    const courseColor = 'green'
    const course = [ template.courseModel() ]
    const pages = [ template.pageModel() ]
    httpCache.handle('GET', 'users/self/colors', { custom_colors: { course_1: courseColor } })
    httpCache.handle('GET', 'courses/1', course)
    httpCache.handle('GET', 'courses/1/pages', { list: pages })
    const tree = shallow(<ConnectedPagesList courseID='1' />)
    expect(tree.find(PagesList).props()).toMatchObject({
      courseColor,
      course,
      pages,
    })
  })

  it('renders empty', () => {
    props.pages = []
    const tree = shallow(<PagesList {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders loading', () => {
    props.course = null
    props.pages = []
    props.isLoading = true
    const tree = shallow(<PagesList {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders front page row', () => {
    props.pages = [template.pageModel({ isFrontPage: true, title: 'Page 1' })]
    const tree = shallow(<PagesList {...props} />)
    const row = diveList(tree.find('FlatList')).find('FeatureRow').first()
    expect(row.prop('title')).toBe('Front Page')
    expect(row.prop('subtitle')).toBe('Page 1')
  })

  it('sorts the pages list', () => {
    const tree = shallow(<PagesList {...props} />)
    const pages = [
      template.pageModel({ title: 'Page B' }),
      template.pageModel({ isFrontPage: true, title: 'Page 1' }),
      template.pageModel({ title: 'Page A' }),
    ]
    tree.setProps({ pages })
    const sorted = tree.find('FlatList').prop('data')
    expect(sorted).toEqual([
      template.pageModel({ isFrontPage: true, title: 'Page 1' }),
      template.pageModel({ title: 'Page A' }),
      template.pageModel({ title: 'Page B' }),
    ])
    tree.setProps({ pages }) // not changed
    expect(tree.find('FlatList').prop('data')).toBe(sorted)
  })

  it('shows access icon for teachers', () => {
    app.setCurrentApp('teacher')
    const tree = shallow(<PagesList {...props} />)
    const row = diveList(tree.find('FlatList')).find('Row').first()
    expect(row.prop('renderImage')()).toMatchSnapshot()
  })

  it('alerts on new loadError', () => {
    const tree = shallow(<PagesList {...props} />)
    tree.setProps({ loadError: null })
    expect(alertError).not.toHaveBeenCalled()
    const loadError = new Error()
    tree.setProps({ loadError })
    expect(alertError).toHaveBeenCalledWith(loadError)
  })

  it('navigates to page details', () => {
    props.navigator = template.navigator({ show: jest.fn() })
    props.pages = [template.pageModel({ url: 'abc' })]
    const tree = shallow(<PagesList {...props} />)
    diveList(tree.find('FlatList')).find('Row').first()
      .simulate('Press', 'abc')
    expect(props.navigator.show).toHaveBeenCalledWith(
      '/courses/1/pages/abc',
      { modal: false }
    )
  })

  it('navigates to new page', () => {
    props.navigator = template.navigator({ show: jest.fn() })
    const tree = shallow(<PagesList {...props} />)
    tree.find('Screen').prop('rightBarButtons')
      .find(({ testID }) => testID === 'pages.list.add.button')
      .action()
    expect(props.navigator.show).toHaveBeenCalledWith(
      '/courses/1/pages/new',
      { modal: true, modalPresentationStyle: 'formsheet' }
    )
  })

  it('shows add button if permitted', () => {
    app.setCurrentApp('teacher')
    const tree = shallow(<PagesList {...props} />)
    expect(
      tree.prop('rightBarButtons').find(({ testID }) => testID === 'pages.list.add.button')
    ).toBeTruthy()
  })

  it('hides add button if not permitted', () => {
    app.setCurrentApp('student')
    const tree = shallow(<PagesList {...props} />)
    expect(tree.prop('rightBarButtons')).toBeFalsy()
  })

  it('shows front page', () => {
    props.navigator = template.navigator({
      traitCollection: (callback) => {
        callback({
          screen: {
            horizontal: 'regular',
          },
          window: {
            horizontal: 'regular',
          },
        })
      },
    })
    props.pages = [template.pageModel({ url: 'front-page', isFrontPage: true })]
    shallow(<PagesList {...props} />)
    expect(props.navigator.show).toHaveBeenLastCalledWith(
      '/courses/1/pages/front-page',
      { modal: false },
    )
  })

  it('does not show front page on deep link modal', () => {
    props.navigator = template.navigator({
      traitCollection: (callback) => {
        callback({
          screen: {
            horizontal: 'regular',
          },
          window: {
            horizontal: 'regular',
          },
        })
      },
      isModal: true,
    })
    props.pages = [template.pageModel({ url: 'front-page', isFrontPage: true })]
    shallow(<PagesList {...props} />)
    expect(props.navigator.show).not.toHaveBeenCalled()
  })

  it('shows placeholder if no front page', () => {
    props.navigator = template.navigator({
      traitCollection: (callback) => {
        callback({
          screen: {
            horizontal: 'regular',
          },
          window: {
            horizontal: 'regular',
          },
        })
      },
    })
    props.pages = [template.pageModel({ isFrontPage: false })]
    const course = props.course = template.courseModel({ id: '1' })
    shallow(<PagesList {...props} />)
    expect(props.navigator.show).toHaveBeenLastCalledWith(
      '/courses/1/placeholder',
      {},
      { courseColor: 'green', course: course.raw },
    )
  })

  it('shows front page or placeholder when switching to split screen', () => {
    let traits = {
      screen: { horizontal: 'compact' },
      window: { horizontal: 'compact' },
    }
    props.navigator = template.navigator({
      traitCollection: (callback) => { callback(traits) },
    })
    props.pages = [template.pageModel({ isFrontPage: false })]
    const course = props.course = template.courseModel({ id: '1' })
    const tree = shallow(<PagesList {...props} />)
    traits = {
      screen: { horizontal: 'regular' },
      window: { horizontal: 'regular' },
    }
    tree.find('Screen').simulate('TraitCollectionChange')
    expect(props.navigator.show).toHaveBeenLastCalledWith(
      '/courses/1/placeholder',
      {},
      { courseColor: 'green', course: course.raw },
    )
  })
})
