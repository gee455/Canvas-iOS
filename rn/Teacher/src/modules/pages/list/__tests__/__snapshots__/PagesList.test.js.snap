// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagesList renders empty 1`] = `
<Screen
  navBarColor="green"
  navBarStyle="dark"
  onTraitCollectionChange={[Function]}
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "New Page",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/Add.png",
        },
        "testID": "pages.list.add.button",
      },
    ]
  }
  showDismissButton={true}
  statusBarStyle="default"
  subtitle="Learn React Native"
  title="Pages"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <FlatList
      ListEmptyComponent={
        <ListEmptyComponent
          title="There are no pages to display."
        />
      }
      data={Array []}
      disableVirtualization={false}
      extraData={
        Object {
          "pages": Array [],
          "selectedPageURL": null,
          "windowTraits": Object {
            "horizontal": "compact",
            "vertical": "regular",
          },
        }
      }
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      onRefresh={[Function]}
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      testID="pages.list.list"
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
</Screen>
`;

exports[`PagesList renders loading 1`] = `
<Screen
  navBarColor="green"
  navBarStyle="dark"
  onTraitCollectionChange={[Function]}
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "New Page",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/Add.png",
        },
        "testID": "pages.list.add.button",
      },
    ]
  }
  showDismissButton={true}
  statusBarStyle="default"
  title="Pages"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <FlatList
      ListEmptyComponent={null}
      data={Array []}
      disableVirtualization={false}
      extraData={
        Object {
          "pages": Array [],
          "selectedPageURL": null,
          "windowTraits": Object {
            "horizontal": "compact",
            "vertical": "regular",
          },
        }
      }
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      onRefresh={[Function]}
      refreshing={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      testID="pages.list.list"
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
</Screen>
`;

exports[`PagesList shows access icon for teachers 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-start",
    }
  }
>
  <AccessIcon
    disableAppSpecificChecks={false}
    entry={
      Object {
        "body": "<p>Hello, World!</p>",
        "created_at": "2017-03-17T19:15:25Z",
        "editing_roles": "teachers",
        "front_page": false,
        "hide_from_students": false,
        "html_url": "/pages/page-1",
        "page_id": "1",
        "published": true,
        "title": "Page 1",
        "updated_at": "2017-03-17T19:15:25Z",
        "url": "page-1",
      }
    }
    image={
      Object {
        "testUri": "../../../src/images/course/Pages.png",
      }
    }
    showAccessIcon={true}
    tintColor="green"
  />
</View>
`;
